packages/grafana-runtime/src/context/GrafanaContext.ts

```diff
@@ -0,0 +1,23 @@
+import React, { useContext } from 'react';
+
+import { GrafanaConfig } from '@grafana/data';
+
+import { LocationService } from '../services/LocationService';
+import { BackendSrv } from '../services/backendSrv';
+
+export interface GrafanaContextType {
+  backend: BackendSrv;
+  location: LocationService;
+  config: GrafanaConfig;
+}
+
+export const GrafanaContext = React.createContext<GrafanaContextType | undefined>(undefined);
+
+export function useGrafana<TExtra extends object = {}>(): GrafanaContextType & TExtra {
```

Comment:
`useGrafana` feels like quite a generic name and I worry people might add all sorts of things to it. If so could that then cause any dependent components to re-render unnecessarily regardless of what they use from this hook? 

If that's the case perhaps splitting these out into separate hooks makes sense?

Reply:
This should never ever cause a re-render. It will have singleton services, those services might have rxjs Subjects react components subscribe to for updates but the hook will never cause a re-render (as it has no state)

This is a pattern I have seen others use. Think this name makes sense from plugin/platform perspective

Reply:
I had similar thought about `useGrafana` name - was thinking about something a bit more verbose like simply `useServices`/`useServicesInternal`.

Regarding the possible re-renders which might be accidentally added later - I think the refs approach I suggest in other comments would make us safe from accidentally introducing context property that could cause issues.

Reply:
@dprokop not sure the useRefs add anything, they don't protected us against re-renders. We just need to make sure we do not add useState & useEffect stuff to this hook : )